src/java/htsjdk/samtools/SAMFileReader.java

```diff
@@ -708,9 +708,11 @@ private boolean isBAMFile(final InputStream stream)
         final byte[] buffer = new byte[buffSize];
         readBytes(stream, buffer, 0, buffSize);
         stream.reset();
-        final byte[] magicBuf = new byte[4];
-        final int magicLength = readBytes(new BlockCompressedInputStream(new ByteArrayInputStream(buffer)), magicBuf, 0, 4);
-        return magicLength == BAMFileConstants.BAM_MAGIC.length && Arrays.equals(BAMFileConstants.BAM_MAGIC, magicBuf);
+        try(final BlockCompressedInputStream bcis = new BlockCompressedInputStream(new ByteArrayInputStream(buffer))){
```

Comment:
Your PR has made some code duplication painfully obvious ;)  Could this entire method just delegate to `SamStreams.isBAMfile()`?